### PR TITLE
Give all pattern synonyms in SmartConstructors.hs type signatures

### PR DESCRIPTION
**Pull request by u--**
in return for uglifying PatternLens

###### Reviewer checklist

- [x] Test coverage: `stack test --coverage`
- [x] Public API documentation: `stack haddock`
- [x] Style conformance

---

### CODE REVIEW

File: src/main/haskell/language-kore/src/Data/Kore/ASTUtils/Substitution.hs [hunk 1]
```diff
@@ -49,78 +48,78 @@ import           Data.Kore.ASTUtils.SmartConstructors
...
+    Forall_ s1 v p -> handleBinder old new Forall_ s1 v p
+    Exists_ s1 v p -> handleBinder old new Exists_ s1 v p
+    pat
      | pat == old -> new
-     | otherwise  -> Fix $ fmap (subst old new) $ unFix pat 
+     | otherwise  -> Fix $ fmap (subst old new) $ unFix pat
 
 -- subst2 a b = cata (\pat -> if Fix pat == a then b else Fix pat)
 
 -- handleBinder
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/ASTUtils/Substitution.hs]: Why not removing all this commented out code?

Comment (traiansf) [reply src/main/haskell/language-kore/src/Data/Kore/ASTUtils/Substitution.hs]: Also, find an editor switch to auto-remove trailing spaces :-)

File: src/main/haskell/language-kore/src/Data/Kore/ASTUtils/SmartConstructors.hs [hunk 2]
```diff
@@ -258,23 +270,27 @@ patternLens
   And_       s2   a b -> And_      <$>          o s2           <*> c a <*> c b
   Bottom_    s2       -> Bottom_   <$>          o s2
   Ceil_   s1 s2   a   -> Ceil_     <$> i s1 <*> o s2           <*> c a
-  DV_        s2   a   -> DV_       <$>          o s2           <*> c a
+  -- DV_        s2   a   -> DV_       <$>          o s2           <*> c a
+  Fix (DomainValuePattern (DomainValue s2 a )) -> DV_ <$> o s2 <*> c a
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/ASTUtils/SmartConstructors.hs]: It's actually not that ugly, though... Jut a little bit ugly

File: src/main/haskell/language-kore/src/Data/Kore/ASTUtils/SmartConstructors.hs [hunk 3]
```diff
@@ -208,13 +220,13 @@ pattern Var_
     :: var level
     -> PureMLPattern level var
 
--- pattern StringLiteral_
---   :: StringLiteral
---   -> Fix (Pattern level var)
+pattern StringLiteral_
+  :: StringLiteral
+  -> PureMLPattern Meta var
```
Comment (traiansf) [inline src/main/haskell/language-kore/src/Data/Kore/ASTUtils/SmartConstructors.hs]: Why doesn't StringLiteral_ and CharLiteral_ match direclty the Sting / Char they wrap?

**Comment by u--** [comment]
I think we should waive the "1 approving review" requirement for really small fixes. For example, xiaohong submitted a PR to fix a typo in semantics-of-k.pdf and it sat there for a week before getting merged.

**Comment by traiansf** [comment]
Jenkins: test this please.

**Comment by virgil-serbanuta** [comment]
About Xiaohong's change - did he assign a reviewer?

**Comment by u--** [comment]
He forgot to assign a reviewer for 4 days, and then after that he forgot to merge it, so I pushed the button. But I'm sure he's busy and he's not gonna spend that long looking after a PR consisting of a 30 second find-and-replace. 

Oftentimes I come across really small things I'd like to fix, but I don't want to make an entire PR for them or put it in another PR.

**Pull request merged**